### PR TITLE
Add keyword search. Make search API work.

### PR DESCRIPTION
**Pull request by khandelwal**
Keywords were not being searched in the backend search that we have. They were in the autocomplete that we had previously, but that's replaced with a simpler autocomplete. 

This adds the keyword search back in. 

It also makes the API for the list of agencies take a query parameter.

### CODE REVIEW

**Comment by khandelwal** [comment]
Fixes: #411

File: docs/source/api.rst [hunk 1]
```diff
@@ -53,6 +53,15 @@ The information returned for each entity is like the following::
       ]
     }
 
+""""""""""""""""""""""""""""""
+GET /api/agency/query='privacy'
```
Comment (ramirezg) [inline docs/source/api.rst]: I think this should be `api/agency/?query=privacy` the documentation query doesn't work.

Comment (khandelwal) [inline docs/source/api.rst]: I have no idea what I was thinking.

**Comment by khandelwal** [comment]
@ramirezg Documentation is now fixed.

**Comment by ramirezg** [comment]
works perfectly and looks good to me

**Pull request merged**